analyzer/core/src/main/scala/io/scalac/newspaper/analyzer/core/Analyzer.scala

```diff
@@ -0,0 +1,53 @@
+package io.scalac.newspaper.analyzer.core
+
+import scala.collection.mutable
+
+class Analyzer {
+
+  import Analyzer._
+
+  private val archive: mutable.Map[String, PreprocessedContent] = new mutable.HashMap()
```

Comment:
If you can hide this behind a trait/interface then you can later replace it with ease (assuming that you want to use a DB later on)